### PR TITLE
7.x islandora scholar 1880

### PR DESCRIPTION
**Pull request by nmader**
JIRA Ticket: https://jira.duraspace.org/browse/ISLANDORA-1880

What does this Pull Request do?

Remove Travis-CI Drupal Test invocations made via Drush, which are being deprecated, as there are no tests in this repository. Updates the YAML to have a MySQL timeout so that long running tests don't cause the MySQL session to timeout. Lastly, adds sudo for the YAML as old builds were grandfathered.

What's new?

Removed the from Drush test runner script fro .travis.yml file. Added Sudo in .travis.yml file as it required for the scrips to run, repos before 2015 were grandfathered in. However, this isn't the case for forks. Also, added a before_script in travis.yml file to extend mysql wait time to 1200 as the builds were failing in Travic-ci.org.

Interested parties

@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by bryjbrown** [comment]
I'd normally be happy to do a review, but Travis stuff is a bit outside of my wheelhouse so I'll defer to someone else and merge on approval.

**Review by DonRichards — APPROVED** [review]
Yes, these are helpful changes. And thanks for pointing to resources in Jira. It makes verification easier. I'm not familiar with the sql ".wait__timeout" part. Your thoughts @bryjbrown? Otherwise it looks good. I think this sql setting works. I'm just not familiar with it

**Comment by bryjbrown** [comment]
@DonRichards Eh... I suppose some Travis tests might go red if the database runs slow, and this would give it extra time to finish up? If we're just increasing the default timeout time, then it can't hurt I suppose. Since these are just Travis settings (which may be revamped in the near future anyway due to lots of code formatting duplication discovered by @DiegoPino in https://github.com/Islandora/islandora_scholar/pull/255#issuecomment-282766805. 

Anyways, merging.

**Comment by bryjbrown** [comment]
@DiegoPino I took a look at the build error you mentioned on IRC. It looks like somehow the last commit in this PR [passed](https://travis-ci.org/Islandora/islandora_scholar/builds/207396504), yet the commit to merge it failed. Very curious because these two should be the exact same codebase, right?

The exact line at the end where the failure occurs says `The command "phpunit sites/all/modules/islandora_scholar/modules/citeproc/tests/CSL_Dateparser.test" exited with 255.`

Niether the `citeproc/includes/CSLDateparser.php` file nor the `citeproc/tests/CSL_Dateparser.test` file have been changed in months, and the first job to run PHP 7 actually errored out instead of failing (timeout), and the following jobs failed with the same error message/stack trace:
```

PHP Fatal error:  Class 'PHPUnit_Framework_TestCase' not found in /home/travis/build/Islandora/islandora_scholar/modules/citeproc/tests/CSL_Dateparser.test on line 14
PHP Stack trace:
PHP   1. {main}() /home/travis/.phpenv/versions/7.0.16/bin/phpunit:0
PHP   2. PHPUnit\TextUI\Command::main() /home/travis/.phpenv/versions/7.0.16/bin/phpunit:465
PHP   3. PHPUnit\TextUI\Command->run() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/TextUI/Command.php:136
PHP   4. PHPUnit\Runner\BaseTestRunner->getTest() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/TextUI/Command.php:159
PHP   5. PHPUnit\Runner\BaseTestRunner->loadSuiteClass() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/Runner/BaseTestRunner.php:71
PHP   6. PHPUnit\Runner\StandardTestSuiteLoader->load() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/Runner/BaseTestRunner.php:130
PHP   7. PHPUnit\Util\Fileloader::checkAndLoad() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/Runner/StandardTestSuiteLoader.php:43
PHP   8. PHPUnit\Util\Fileloader::load() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/Util/Fileloader.php:39
PHP   9. include_once() phar:///home/travis/.phpenv/versions/7.0.16/bin/phpunit/phpunit/Util/Fileloader.php:55
```

My guess is that something went wrong with the installation of phpunit (maybe the host went down temporarily?) so its not that the test is failing, but instead that the test couldn't be run inside the build environment.

Is there any way we trigger a rebuild and run the tests again? I really can't think of any reason why the last commit of the PR would pass but the merge commit would fail other than a network issue inside Travis.

**Comment by DiegoPino** [comment]
@bryjbrown++ thanks! Ok, will give it a try again. Not sure why it failed but your logic makes sense. Job `#550`, which belongs to this you merge passed ok, but [551](https://travis-ci.org/Islandora/islandora_scholar/builds/210380956) is failing (not sure why there is a 551? since it seems to be newer and points to the same pull request... probably it's the Travis-CI run that triggers on merge, v/s the one that triggers on pull...). Restarted all PHP 7 and 7.1 on those without luck. Will give this a try again tomorrow morning with a different strategy. If something is wrong with this pull, then we will know for sure on the next pull to Scholar. Pinged you mostly to ask if this is a new issue or a Travis-CI weirdness.

**Comment by bryjbrown** [comment]
@DiegoPino Definitely a new issue to my eyes... I'm not a Travis expert so I was deferring to others on approving the changes, so if theres something sour in this new .travis.yml setup we can always just roll the commit back.

**Comment by DiegoPino** [comment]
@bryjbrown did my homework and well, we will have some issues (not in scholar but everywhere we test with php7)

It all began with some googling until:

https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-6.0.md#changed-1
**(PHPUnit's units of code are now namespaced!)** 😱 

So things like \PHPUnit_Framework_TestCase need to be moved to \PHPUnit\Framework\TestCase probably. 

You will ask then "Diego, why other PHP versions don't fail, are you sure?", to what I will have to answer 
"Bryan, I'm never sure, that is for sure, but seems like older PHP versions run with an older PHPUnit version, but 7 needs the most recent"

Big questions is "how do we proceed from here?". Probably an autoloader and making sure we run even php5 with the newest PHPUnit. A lot of PHP7 test code was pushed lately...!!
probably a talk for today's release call...

Thanks!

**Comment by bryjbrown** [comment]
Solid detective work, @DiegoPino. Looking forward to today's call. :+1:

**Comment by DiegoPino** [comment]
@bryjbrown opened a JIRA ticket, https://jira.duraspace.org/browse/ISLANDORA-1935. Affects this module (sorry) and Tuque.

**Pull request merged**
